crates/nuts/src/amount.rs

```diff
@@ -234,6 +234,18 @@ impl std::ops::Div for Amount {
     }
 }
 
+impl PartialOrd<u64> for Amount {
+    fn partial_cmp(&self, other: &u64) -> Option<Ordering> {
+        Some(self.0.cmp(other))
+    }
+}
+
+impl PartialEq<u64> for Amount {
+    fn eq(&self, other: &u64) -> bool {
+        self.0 == *other
+    }
+}
+
```

Comment:
not a big fan of that. Can't you just do the conversion instead? and then compare the converted values. 

Or if your really don't want to. I would use `AsRef` and use this instead.

But for u64 conversion are free in term of execution, so you can just use those as much as you want